collie/interactions/datasets.py

```diff
@@ -313,10 +319,13 @@ def __init__(self,
         if seed is None:
             seed = collie.utils.get_random_seed()
 
+        self.negative_sample_type = negative_sample_type
```

Comment:
Should we do some validation on `negative_sample_type` to make sure it is one of the expected values?